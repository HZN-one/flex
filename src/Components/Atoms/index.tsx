export * from "./FAButton";
export * from "./Icon";
export * from "../Molecules/FMFormTextField";
export * from "./FATextField";
export * from "../Molecules/FMSelect";
export * from "./FACheckbox";
export * from "./FASwitch";
export * from "./FARadio";
export * from "./FAPagination";
export * from "./FAChip";
export * from "./FATooltip";
export * from "./FAFormControlLabel";
export * from "./FAFormControl";
