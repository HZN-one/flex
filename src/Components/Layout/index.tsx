export * from "./Initiator";
export * from "./FLSettingsNavbar";
