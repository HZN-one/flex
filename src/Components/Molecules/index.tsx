export * from "./FMAccordion";
export * from "./FMSearch";
export * from "./FMCard";
export * from "./FMStepper";
export * from "./FMTableSkeleton";
export * from "./FMTimeline";
export * from "./FMLabelValue";
export * from "./FMTabsMenu";
export * from "./FMInputAvatar";
