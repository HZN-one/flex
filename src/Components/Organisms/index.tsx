export * from "./FOCardWithImage";
export * from "./FOCardWithStepper";
export * from "./FOCardWithTextField";
export * from "./FODialogWithAlert";
export * from "./FOHeader";
export * from "./FOPaperRow";
export * from "./FOSideBar";
