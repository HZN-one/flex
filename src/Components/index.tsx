export * from './Atoms'
export * from './Layout'
