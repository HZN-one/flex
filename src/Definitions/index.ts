export * from "./Theme";
export * from "./constants";
export * from "./helpers";
