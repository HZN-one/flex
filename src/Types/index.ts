export type ElementPosition = "start" | "center" | "end";
