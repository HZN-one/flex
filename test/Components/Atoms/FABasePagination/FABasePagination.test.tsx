/**
 * @jest-environment jsdom
 */

 import * as React from 'react'
 import '@testing-library/jest-dom'
 
 import { FABasePagination } from '../../../../src/Components'
 
 import { fireEvent, render } from '@testing-library/react'
 
 describe('FABasePagination', () => {
   it('should match snapshot', () => {
     const { container } = render(
       <FABasePagination testID="pagination-example" />
     )
     expect(container).toMatchSnapshot()
   })
 })
 