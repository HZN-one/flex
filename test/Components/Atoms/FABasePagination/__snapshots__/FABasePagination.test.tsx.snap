// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FAPagination should match snapshot 1`] = `
<div>
  <div
    class="css-1p5q5e5-MuiStack-root"
  >
    <nav
      aria-label="pagination navigation"
      class="MuiPagination-root MuiPagination-text css-1oj2twp-MuiPagination-root"
      data-testid="pagination-example"
    >
      <ul
        class="MuiPagination-ul css-wjh20t-MuiPagination-ul"
      >
        <li>
          <button
            aria-label="Go to previous page"
            class="MuiButtonBase-root Mui-disabled MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular Mui-disabled MuiPaginationItem-previousNext css-yuzg60-MuiButtonBase-root-MuiPaginationItem-root"
            disabled=""
            tabindex="-1"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-g2z002-MuiSvgIcon-root-MuiPaginationItem-icon"
              data-testid="NavigateBeforeIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
              />
            </svg>
          </button>
        </li>
        <li>
          <button
            aria-current="true"
            aria-label="page 1"
            class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular Mui-selected MuiPaginationItem-page css-yuzg60-MuiButtonBase-root-MuiPaginationItem-root"
            tabindex="0"
            type="button"
          >
            1
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </li>
        <li>
          <button
            aria-label="Go to page 2"
            class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-page css-yuzg60-MuiButtonBase-root-MuiPaginationItem-root"
            tabindex="0"
            type="button"
          >
            2
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </li>
        <li>
          <button
            aria-label="Go to page 3"
            class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-page css-yuzg60-MuiButtonBase-root-MuiPaginationItem-root"
            tabindex="0"
            type="button"
          >
            3
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </li>
        <li>
          <button
            aria-label="Go to page 4"
            class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-page css-yuzg60-MuiButtonBase-root-MuiPaginationItem-root"
            tabindex="0"
            type="button"
          >
            4
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </li>
        <li>
          <button
            aria-label="Go to page 5"
            class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-page css-yuzg60-MuiButtonBase-root-MuiPaginationItem-root"
            tabindex="0"
            type="button"
          >
            5
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </li>
        <li>
          <div
            class="MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-ellipsis css-1v2lvtn-MuiPaginationItem-root"
          >
            â€¦
          </div>
        </li>
        <li>
          <button
            aria-label="Go to page 10"
            class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-page css-yuzg60-MuiButtonBase-root-MuiPaginationItem-root"
            tabindex="0"
            type="button"
          >
            10
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </li>
        <li>
          <button
            aria-label="Go to next page"
            class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-text MuiPaginationItem-circular MuiPaginationItem-previousNext css-yuzg60-MuiButtonBase-root-MuiPaginationItem-root"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-g2z002-MuiSvgIcon-root-MuiPaginationItem-icon"
              data-testid="NavigateNextIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </li>
      </ul>
    </nav>
  </div>
</div>
`;
